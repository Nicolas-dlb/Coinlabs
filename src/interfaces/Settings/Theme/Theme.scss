.settings_theme {
  width: 85%;
  align-self: flex-end;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 20px;
  border-radius: 15px;
  background: var(--third-bg-color);
  margin-top: 2vh;
  .settings_theme_container {
    display: flex;
    align-items: center;
    justify-content: center;
    margin-top: 2vh;
    .btn_theme_active {
      background: var(--primary-font-color) !important;
      color: black !important;
    }
    .btn-theme {
      background: #2b263f;
      padding: 8px 14px;
      border-radius: 5px;
      border: none;
      color: var(--primary-font-color);
      margin-right: 15px;
      cursor: pointer;
    }
  }
}
