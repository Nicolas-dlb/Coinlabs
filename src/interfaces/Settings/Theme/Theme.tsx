import React, { useState } from "react";
import "./Theme.scss";

function Theme() {
  const [activeTheme, setActiveTheme] = useState("Dark");
  return (
    <div className="settings_theme">
      <h3>Theme</h3>
      <div className="settings_theme_container">
        <button
          onClick={() => setActiveTheme("Dark")}
          type="button"
          className={
            activeTheme === "Dark" ? "btn-theme btn_theme_active" : "btn-theme"
          }
        >
          Dark
        </button>
        <button
          onClick={() => setActiveTheme("Grey")}
          type="button"
          className={
            activeTheme === "Grey" ? "btn-theme btn_theme_active" : "btn-theme"
          }
        >
          Grey
        </button>
      </div>
    </div>
  );
}

export default Theme;
