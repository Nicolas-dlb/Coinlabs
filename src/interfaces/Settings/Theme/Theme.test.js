import React from "react";
import { render } from "enzyme";
import { act } from "react-dom/test-utils";
import Theme from "./Theme";

describe("Theme", () => {
  let ThemeComponent;
  beforeEach(async () => {
    await act(async () => {
      ThemeComponent = render(<Theme />);
    });
  });
  it("should render correctly", () => {
    expect(ThemeComponent).toMatchSnapshot();
  });

  it("should render Theme title", () => {
    expect(ThemeComponent.find("h3").text()).toBe("Theme");
  });

  it("should render 2 button with Dark and Grey text", () => {
    expect(ThemeComponent.find("button").length).toBe(2);
    expect(ThemeComponent.find("button").first().text()).toBe("Dark");
    expect(ThemeComponent.find("button").last().text()).toBe("Grey");
  });
});
