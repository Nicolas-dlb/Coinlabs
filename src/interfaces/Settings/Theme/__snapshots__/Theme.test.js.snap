// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme should render correctly 1`] = `
<div
  class="settings_theme"
>
  <h3>
    Theme
  </h3>
  <div
    class="settings_theme_container"
  >
    <button
      class="btn-theme btn_theme_active"
      type="button"
    >
      Dark
    </button>
    <button
      class="btn-theme"
      type="button"
    >
      Grey
    </button>
  </div>
</div>
`;
